tests/dataset_runner.py

```diff
@@ -149,28 +166,82 @@ def complete_submission(self):
             raise RuntimeError(f"PUT {submit_url} returned {response.status_code}: {response.content}")
         Progress.report("  done.\n")
 
-    def wait_for_primary_and_results_bundles(self):
+    def wait_for_primary_bundles(self):
         """
         We used to wait for all primary bundles to be created before starting to look for results.
         It appears that with large submissions, results can start to appear before bundle export is finished,
         so we now monitor both kinds of bundles simultaneously.
         """
-        Progress.report("WAITING FOR PRIMARY AND RESULTS BUNDLE(s) TO BE CREATED...")
+        Progress.report("WAITING FOR PRIMARY BUNDLE(s) TO BE CREATED...")
         self.expected_bundle_count = self.dataset.config["expected_bundle_count"]
         WaitFor(
-            self._count_primary_and_secondary_bundles
+            self._count_primary_bundles_and_report
         ).to_return_value(value=self.expected_bundle_count)
 
-    def _count_primary_and_secondary_bundles(self):
+    def wait_for_analysis_workflows(self):
+        if not self.analysis_agent:
+            Progress.report("NO CREDENTIALS PROVIDED FOR ANALYSIS AGENT, SKIPPING WORKFLOW(s) CHECK...")
+        else:
+            Progress.report("WAITING FOR ANALYSIS WORKFLOW(s) TO FINISH...")
+            WaitFor(
+                self._count_analysis_workflows_and_report
+            ).to_return_value(value=self.expected_bundle_count)
+            
+    def _count_analysis_workflows_and_report(self):
+        if self._analysis_workflows_count() < self.expected_bundle_count:
+            self._count_analysis_workflows()
+        Progress.report("  successful analysis workflows: {}/{}".format(
+            self._analysis_workflows_count(),
+            self.expected_bundle_count
+        ))
+        return self._analysis_workflows_count()
+
+    def _count_analysis_workflows(self):
+        for bundle_uuid in self.submission_envelope.bundles():
```

Comment:
To get workflow info in the scale test, I think it is more efficient to make a single request to query for workflows with the `project_shortname` label instead of a request per bundle. You may have to set `with_labels` to `False` to get around the issue you linked in Cromwell..so you would not be able to log which workflows failed for which bundle but you could give still provide summary info.